unity/Assets/Scripts/AgentManager.cs

```diff
@@ -456,24 +458,46 @@ private enum serverTypes { WSGI, FIFO };
             );
         }
 
-        //note if updating the primary agent's main camera, this allows it to be detached! use at your own risk
-        if(!agentPositionRelativeCoordinates) {
-            // update the position and rotation and keep camera in world space coordinates
-            camera.transform.SetParent(null);
-            camera.gameObject.transform.position = position;
-            camera.gameObject.transform.eulerAngles = rotation;
+        if (parent != null && parent != "agent" && parent != "world") {
+            throw new InvalidOperationException(
+                $"parent: {parent} must be one of: null, agent, or world."
+            );
+        }
+
+        var agent = this.agents[agentId];
+
+        if (agentPositionRelativeCoordinates) {
+            Transform oldParent = camera.transform.parent;
+            camera.transform.SetParent(agent.transform);
+            camera.transform.localScale = Vector3.one; // Previous comment suggested that these local scale changes are necessary after reparenting
+
+            if (position.HasValue) {
+                camera.transform.localPosition = position.Value;
+            }
+            if (rotation.HasValue) {
+                camera.transform.localEulerAngles = rotation.Value;
+            }
+            camera.transform.SetParent(oldParent);
+            camera.transform.localScale = Vector3.one;
         } else {
-            //makes sure camera is child of agent, and reposition in agent relative space
-            if(camera.transform.parent != primaryAgent.transform) {
-                camera.transform.SetParent(primaryAgent.transform);
-                //for some reason reparenting sometimes makes the scale slightly off...
-                //so we are gonna force it to be uniform just in case cause floats suck
-                camera.transform.localScale = Vector3.one;
+            if (position.HasValue) {
+                camera.gameObject.transform.position = position.Value;
```

Comment:
This is fine but a thing to consider: there are a lot of if else statements that end with the same code like assigning the position. 
I may try refactoring so that only those lines exist once so that is easier to debug, as much as possible.

You can try using the `??=` operator to optionally (assign a value that is null)[https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator#examples]